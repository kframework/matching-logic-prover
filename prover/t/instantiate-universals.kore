sort A
symbol a() : A
axiom functional(a)
symbol f(Int,Int) : Int
claim \implies(
  \forall{} \forall{X{Int}, Y{A}} \equals(f(X{Int},Y{A}), 0),
  \equals(f(3, a()), 0)
)
strategy intros H
       . instantiate-universals(in: H, vars: X,Y, with: 3,a())
       . apply-equation -> H at 0 by []
       . reflexivity
