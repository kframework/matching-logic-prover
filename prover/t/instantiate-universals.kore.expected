<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom ax0 : functional ( a )
    </declaration> <declaration>
      axiom ax1 : \implies ( \forall { .Patterns } \forall { X { Int } , Y { A } , .Patterns } \equals ( f ( X { Int } , Y { A } , .Patterns ) , 0 ) , \equals ( f ( 3 , a ( .Patterns ) , .Patterns ) , 0 ) )
    </declaration> <declaration>
      sort A
    </declaration> <declaration>
      symbol a ( .Sorts ) : A
    </declaration> <declaration>
      symbol f ( Int , Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
